### PR TITLE
Feature/comment section

### PR DESCRIPTION
**Pull request by developer-mindpath**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Added new comment section components: `CommentsUpvotes`, `WordpressComments`, `CommentsSolidBackground`, and `DefaultCommentSection`.
- Implemented form handling and validation using `react-hook-form` and `zod`.
- Included UI elements like buttons, text areas, and popovers in the new components.
- Rendered lists of comments with various functionalities such as upvotes, likes, and replies.
- Added Storybook stories for the new comment section components.
- Updated Storybook configuration to include new story paths.
- Updated type definition for `check` function in `AdvancedDropdownFilterKeywords`.
- Added a newline at the end of `.storybook/manager-head.html`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>comment-upvotes.tsx</strong><dd><code>Add CommentsUpvotes component with upvote functionality</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx

<li>Added a new <code>CommentsUpvotes</code> component with upvote functionality.<br> <li> Implemented form handling using <code>react-hook-form</code> and <code>zod</code>.<br> <li> Included UI elements like buttons, text areas, and popovers.<br> <li> Rendered a list of comments with upvote and reply options.<br>

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-e5eb6f4c2bc3ce0ab53817eb7b7f8eca1fa68322a881f26c1b954af2df7c3ce6">+320/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>wordpress-comments.tsx</strong><dd><code>Add WordpressComments component with form validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/wordpress-comments.tsx

<li>Added a new <code>WordpressComments</code> component.<br> <li> Implemented form handling with validation using <code>react-hook-form</code> and <br><code>zod</code>.<br> <li> Included UI elements like buttons, text areas, and popovers.<br> <li> Rendered a list of comments with like and reply options.<br>

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-4dbb23be8b9f66135cd4aeae1c1301461a47f324d9808dbcfd3b656571f5a975">+323/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>solid-background.tsx</strong><dd><code>Add CommentsSolidBackground component with form handling</code>&nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/solid-background.tsx

<li>Added a new <code>CommentsSolidBackground</code> component.<br> <li> Implemented form handling using <code>react-hook-form</code> and <code>zod</code>.<br> <li> Included UI elements like buttons, text areas, and popovers.<br> <li> Rendered a list of comments with like and reply options.<br>

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-544e3166192ad2c3908364384e3195966123b66465085b98a87195b0ae787dd5">+263/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>default.tsx</strong><dd><code>Add DefaultCommentSection component with form handling</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/default.tsx

<li>Added a new <code>DefaultCommentSection</code> component.<br> <li> Implemented form handling using <code>react-hook-form</code> and <code>zod</code>.<br> <li> Included UI elements like buttons, text areas, and popovers.<br> <li> Rendered a list of comments with reply options.<br>

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-499f4a643a30e8365e0c3ee450a7044208c562ac3be7831f9a952f19b3fe178c">+197/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>advanced-keywords.tsx</strong><dd><code>Update type definition for check function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/application-ui/dropdown-filters/advanced-keywords.tsx

- Updated type definition for `check` function.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-22fdb7b1310f7ad325e64c2b2d255c0f9f0573b55da72904c98e3a74efa28c40">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>main.ts</strong><dd><code>Update Storybook configuration for new story paths</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.storybook/main.ts

- Updated Storybook configuration to include new story paths.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-2d9ab5214985b9a608b4cd47ed561a301cfd4c7d5352c7a94ee046d4e1f1af10">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests
</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>comment-upvotes.stories.tsx</strong><dd><code>Add Storybook story for CommentsUpvotes component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/stories/comment-upvotes.stories.tsx

- Added Storybook story for `CommentsUpvotes` component.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-99949d46251e6c2dd54a11882d9da76d0ae32599162279488010db11a0c89b95">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>solid-background.stories.tsx</strong><dd><code>Add Storybook story for CommentsSolidBackground component</code></dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/stories/solid-background.stories.tsx

- Added Storybook story for `CommentsSolidBackground` component.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-c5aabf077f47897d38f5df50f2cd32a7eb1e755ddda972ed077013beb8194201">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>default.stories.tsx</strong><dd><code>Add Storybook story for DefaultCommentSection component</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/stories/default.stories.tsx

- Added Storybook story for `DefaultCommentSection` component.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-775086e169d327fbfe1f9c4b23859d7549e19ecc27bd4dc533eb55b63807db03">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>wordpress-comments.stories.tsx</strong><dd><code>Add Storybook story for WordpressComments component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/components/sections/publisher-ui/comment-section/stories/wordpress-comments.stories.tsx

- Added Storybook story for `WordpressComments` component.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-c8f623896f2b99de305da13bc45ec19c5a992362a1e7f166500496e34056ec0a">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Formatting
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>manager-head.html</strong><dd><code>Add newline at end of manager-head.html</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.storybook/manager-head.html

- Added a newline at the end of the file.

</details>
    

  </td>
  <td><a href="https://github.com/modernui-io/modernuiweb/pull/10/files#diff-48290ed8357dc288432092c676e90e7b8702929064d7f52964444711b596fbf7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+      <MuiLayout.Container className="mx-auto max-w-2xl px-4">
+        <div className="mb-6 flex items-center justify-between">
+          <Heading level={4} className="font-bold text-foreground lg:text-2xl">
+            Discussion (20)
+          </Heading>
+          <div>
+            <Button
+              type="button"
+              className="rounded-lg border border-primary-200 bg-background px-3 py-2 text-sm font-medium text-foreground hover:bg-primary-100 hover:text-primary-700 focus:z-10 focus:outline-none focus:ring-2 focus:ring-primary-200"
+            >
```
Comment (nnennajohn) [inline src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx:175]: Is this a variant, or can it be a variant

File: src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                        </Button>
+                      ))}
+                    </div>
+                  </div>
+                </div>
+              </div>
+              <Button
+                type="submit"
+                className="inline-flex items-center rounded-lg bg-primary-700 px-4 py-2.5 text-center text-xs font-medium hover:bg-primary-800 focus:ring-4 focus:ring-primary-200"
+              >
```
Comment (nnennajohn) [inline src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx:247]: Please try to use button variants all across

Comment (developer-mindpath) [reply src/components/sections/publisher-ui/comment-section/comment-upvotes.tsx:237]: I have applied the variants where necessary and used the default variant everywhere else. The type="submit" is specifically for the form. Please let me know if any further changes are needed.

**Review by nnennajohn — APPROVED** [review]
Thank you.

**Pull request merged**